### ISSUE TITLE
Add support for TIME[STAMP] WITH TIME ZONE in HSQLDB

### ISSUE THREAD

**Issue by bassmake** [jOOQ/jOOQ#5895]
Hi,
I've been playing around with zone support in HSQL and JOOQ correctly created OffsetDateTime for my `timestamp with time zone` column, but when I tried to insert row into table I've got
```
org.jooq.exception.DataAccessException: SQL [null]; data exception: invalid datetime format

	at org.jooq_3.9.1.HSQLDB.debug(Unknown Source)
	at org.jooq.impl.Tools.translate(Tools.java:1983)
	at org.jooq.impl.AbstractBindContext.bindValue(AbstractBindContext.java:130)
	at org.jooq.impl.Val.accept(Val.java:96)
	at org.jooq.impl.AbstractBindContext.bindInternal(AbstractBindContext.java:257)
	at org.jooq.impl.AbstractBindContext.visit0(AbstractBindContext.java:91)
	at org.jooq.impl.AbstractContext.visit0(AbstractContext.java:402)
	at org.jooq.impl.AbstractContext.visit(AbstractContext.java:168)
	at org.jooq.impl.QueryPartList.accept(QueryPartList.java:104)
	at org.jooq.impl.AbstractBindContext.bindInternal(AbstractBindContext.java:257)
	at org.jooq.impl.AbstractBindContext.visit0(AbstractBindContext.java:91)
	at org.jooq.impl.AbstractContext.visit0(AbstractContext.java:402)
	at org.jooq.impl.AbstractContext.visit(AbstractContext.java:168)
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:345)
	at org.jooq.impl.AbstractDelegatingQuery.execute(AbstractDelegatingQuery.java:135)
	at sk.bsmk.pbl.persistence.CampaignJooqRepository.storeCampaign(CampaignJooqRepository.kt:28)
	at sk.bsmk.pbl.persistence.CampaignJooqRepositoryTest.thatCampaignCanBeStored(CampaignJooqRepositoryTest.java:51)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.springframework.test.context.junit4.statements.RunBeforeTestExecutionCallbacks.evaluate(RunBeforeTestExecutionCallbacks.java:73)
	at org.springframework.test.context.junit4.statements.RunAfterTestExecutionCallbacks.evaluate(RunAfterTestExecutionCallbacks.java:83)
	at org.springframework.test.context.junit4.statements.RunBeforeTestMethodCallbacks.evaluate(RunBeforeTestMethodCallbacks.java:75)
	at org.springframework.test.context.junit4.statements.RunAfterTestMethodCallbacks.evaluate(RunAfterTestMethodCallbacks.java:86)
	at org.springframework.test.context.junit4.statements.SpringRepeat.evaluate(SpringRepeat.java:84)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:250)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:97)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.springframework.test.context.junit4.statements.RunBeforeTestClassCallbacks.evaluate(RunBeforeTestClassCallbacks.java:61)
	at org.springframework.test.context.junit4.statements.RunAfterTestClassCallbacks.evaluate(RunAfterTestClassCallbacks.java:70)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.run(SpringJUnit4ClassRunner.java:189)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:51)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:237)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:147)
Caused by: java.sql.SQLDataException: data exception: invalid datetime format
	at org.hsqldb.jdbc.JDBCUtil.sqlException(Unknown Source)
	at org.hsqldb.jdbc.JDBCUtil.throwError(Unknown Source)
	at org.hsqldb.jdbc.JDBCPreparedStatement.setParameter(Unknown Source)
	at org.hsqldb.jdbc.JDBCPreparedStatement.setString(Unknown Source)
	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.setString(HikariProxyPreparedStatement.java)
	at org.jooq.tools.jdbc.DefaultPreparedStatement.setString(DefaultPreparedStatement.java:138)
	at org.jooq.impl.DefaultBinding.set(DefaultBinding.java:1073)
	at org.jooq.impl.DefaultBindContext.bindValue0(DefaultBindContext.java:62)
	at org.jooq.impl.AbstractBindContext.bindValue(AbstractBindContext.java:127)
	... 49 more
Caused by: org.hsqldb.HsqlException: data exception: invalid datetime format
	at org.hsqldb.error.Error.error(Unknown Source)
	at org.hsqldb.error.Error.error(Unknown Source)
	at org.hsqldb.types.DateTimeType.convertToDatetimeSpecial(Unknown Source)
	at org.hsqldb.types.DateTimeType.convertToType(Unknown Source)
	... 56 more
```

project can be found in https://github.com/bassmake/hi-spring-kotlin
just run `gradlew clean generatePblJooqSchemaSource` and then test in `CampaignJooqRepositoryTest.thatCampaignCanBeStored`

**Comment by sysoutch** [comment]
I have the exactly same problem. Any news about that?

**Comment by lukaseder** [comment]
Thank you very much for the ping, @sysoutch. This seems to have accidentally never received a milestone, which is why it got off radar. I'll look into this right away.

**Comment by lukaseder** [comment]
The workaround for older versions of jOOQ is to use custom data type bindings:
https://www.jooq.org/doc/latest/manual/sql-building/queryparts/custom-bindings

**Comment by lukaseder** [comment]
Implemented for jOOQ 3.12 - All JSR 310 related integration tests pass. I won't merge this to 3.11 as it affects code generation output incompatibly, which may be surprising in a patch release.

Thanks again for reporting, and for the ping!
